File: lib/disks.py
class Disks:
    psutil = None

    def __init__(self, psutil):
        self.psutil = psutil

    def snapshot(self):
        return {
            "disk_usage": self.psutil.disk_usage('/'),
            "disk_partitions": self.psutil.disk_partitions(),
            "disk_io_counters": self.psutil.disk_io_counters()
        }

File: src/collector/Disks.go
package collector

import (
	psutil_disk "github.com/shirou/gopsutil/disk"
)


// Disks is the struct that contains data about the Disks
type Disks struct {
	DiskUsage      interface{} `json:"disk_usage_stat"`
	DiskPartition  interface{} `json:"disk_partition_stat"`
	DiskIOCounters interface{} `json:"disk_io_counters_stat"`
}

// Collect helps to collect data about the Disks
// and store it in the Disks struct
func (DisksPtr *Disks) Collect(disk_partition bool) *Disks {
	DisksPtr.DiskUsage, _ = psutil_disk.DiskUsage("/")
	DisksPtr.DiskIOCounters, _ = psutil_disk.DiskIOCounters()

	if disk_partition {
		DisksPtr.DiskPartition, _ = psutil_disk.DiskPartitions(true)
	}

	return DisksPtr
}
